internal/worker/providerservicefactory/manifold.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package providerservicefactory
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/juju/worker/v4"
+	"github.com/juju/worker/v4/dependency"
+
+	"github.com/juju/juju/core/changestream"
+	coredatabase "github.com/juju/juju/core/database"
+	coremodel "github.com/juju/juju/core/model"
+	domainservicefactory "github.com/juju/juju/domain/servicefactory"
+	"github.com/juju/juju/internal/servicefactory"
+	"github.com/juju/juju/internal/worker/common"
+)
+
+// Logger represents the logging methods called.
+type Logger interface {
+	Tracef(string, ...interface{})
+	Debugf(message string, args ...any)
+	Warningf(message string, args ...any)
+}
+
+// ManifoldConfig holds the information necessary to run a provider service
+// factory worker in a dependency.Engine.
+type ManifoldConfig struct {
+	ChangeStreamName                string
+	Logger                          Logger
+	NewWorker                       func(Config) (worker.Worker, error)
+	NewProviderServiceFactoryGetter ProviderServiceFactoryGetterFn
```

Comment:
Good to comment these here in the in the worker, particularly regarding one being for supply to consumers.